### ISSUE TITLE
solve_sparse_core_many_into: column-major RHS layout prevents inner-loop vectorization

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/feral#57]
## Summary

The supernodal multi-RHS forward/back-substitution in `solve_sparse_core_many_into` (numeric/solve.rs) lays out the working buffer column-major (`w[c*nrow + i]`), which makes the innermost per-RHS update a strided gather rather than a contiguous axpy. The compiler can't auto-vectorize it, and per-RHS cost is roughly equal to looping the single-RHS path — so multi-RHS is a wash on real workloads even though all the structure for amortization is there.

## Where it hurts

Forward-sub loop (similar shape on the back-sub side):

```rust
// numeric/solve.rs, solve_sparse_core_many_into
for j in 0..nelim {
    for i in (j + 1)..nrow {
        let l_ij = ff.l[j * nrow + i];
        for c in 0..nrhs {
            let off = c * nrow;                 // stride = nrow (often 100s)
            w[off + i] -= l_ij * w[off + j];
        }
    }
}
```

The inner `c`-loop reads `w[i], w[nrow+i], w[2*nrow+i], …` — stride-`nrow` access. LLVM won't vectorize this; it sees dependent strided loads and emits scalar code. Single-RHS `solve_sparse_core_into` is contiguous and fast, so feral's `solve_many` essentially serial-loops the same per-pivot cost.

The doc comment near this site (`"compiler auto-vectorizes for small constant or runtime `nrhs`"`) is optimistic; godbolt'ing the loop with nrhs ≥ 64 confirms scalar codegen.

## Measured impact

Driving `Feral` through `pounce`'s `kkt_solve_many` (sparse KKT back-solves, n+m = 510 .. 4094, nrhs = n):

```
                       1-RHS       N-RHS (per-RHS)   ratio
n=256  kkt_dim=510      135 µs       95.2 µs          0.71×
n=512  kkt_dim=1022     204 µs      194.1 µs          0.95×
n=1024 kkt_dim=2046     359 µs      384.0 µs          1.07×
n=2048 kkt_dim=4094     935 µs      764.5 µs          0.82×
```

For comparison, a dense LU + multi-RHS GETRS on the same matrices runs ~5-8× faster per RHS on the same hardware. The supernodal factor structure here is amortizable in principle — we just can't reach it through this layout.

This is the dominant cost for `jax.jacrev` over a `JaxProblem.solve` in pounce#77 (one converged factor, n_rhs cotangents), which is what surfaced it.

## Proposed fixes (in order of effort)

1. **Row-major working buffer inside the supernode loop**: store `w[i*nrhs + c]` instead of `w[c*nrow + i]`. The inner `c`-loop becomes contiguous, LLVM auto-vectorizes, and per-pivot work amortizes across RHS. Probably ~40 lines of patch — only the workspace gather/scatter and the inner loop change; `SolveManyWorkspace` is private so no API impact. Best first try.

2. **BLAS-3 `dtrsm` on each supernode's frontal panel**: feral is already supernodal, panels are dense, this is the textbook supernodal multifrontal multi-RHS optimization. Bigger change but should deliver the 5-10× nrhs speedup at moderate sizes. Optional second step once (1) is measured.

The same change applies to the back-substitution loop in the same function (symmetric structure).

## Repro

The probe lives in `pounce/python` (pounce#77 follow-up). Happy to extract a standalone feral-only benchmark if useful — basically build a random sparse SPD matrix at n=512..4096, factor once, then time `solve_many` at various nrhs and compare against `solve` looped.

## Context

Surfaced via pounce's batched KKT back-solve path (`PdFullSpaceSolver::solve_many_cached` → `TSymLinearSolver::multi_solve` → `FeralSolverInterface::backsolve` → `solver.solve_many`). The pounce-side plumbing is now in place to take advantage of a faster feral multi-RHS path with no further changes.

**Comment by jkitchin** [comment]
Fixed across this session.

**fix #1 (80348f9):** row-major per-supernode `w` buffer so the per-RHS inner loops vectorize.

**fix #2 (9c2c716):** register-blocked TRSM + GEMM panel kernels behind `nrhs >= 32`, plus a row-major internal `y` (gather/scatter becomes a contiguous memcpy — this removed a power-of-two-`n` cache-aliasing regression) and fused forward+D-solve. Per-RHS speedup on 2-D Laplacians: n=484 ~4-5x, n=1024 ~3x, n=2025 ~5-6x; bit-identical forward, parity to machine precision.

The refined path (the default, and pounce's KKT back-solve path) is handled in #58 (closed): batched iterative refinement now goes through the panel kernel too, ~2.5-3x faster than the per-column loop, and the Python `solve_refined(2-D)` binding delivers it (an allocation overhead that masked it was found and removed, 2e096e9).

Verified: lib 317 pass; multi_rhs 13 pass (incl. exact bit-identical bands); clippy --all-targets clean. Benches: `bench_multirhs` (2-D Laplacian) and `bench_dense_multirhs` (dense Hessian). Python notebook `02_multi_rhs_batched` showcases the unrefined (~3x) and refined (~2-3x) batched solves end-to-end. Docs: getting-started, Python bindings page, and the mdBook.
